hack/lib/util/misc.sh

```diff
@@ -233,3 +233,35 @@ function os::util::list_go_src_files() {
 	\) -name '*.go' | sort -u
 }
 readonly -f os::util::list_go_src_files
+
+# os::util::docker calls docker, elevating privileges if necessary.
+# Scripts that expect to need to elevate privilges should call
+# os::util::environment::use_sudo in their preamble.
```

Comment:
How/why is the script supposed to know if it will need to elevate privileges? The script knows it needs to run docker. We should just make that work.